### PR TITLE
[WB-4019]: Add Line series plot for custom charts

### PR DESCRIPTION
**Pull request by KyleGoyette**
Link to jira issue:
https://wandb.atlassian.net/browse/WB-4019

The plot spec:
```{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "description": "A plot for an arbitrary number of lines",
  "data": {
    "name": "wandb"
  },
  "title": "${string:title}",
  "layer": [
    {
      "mark": {"type": "line", "interpolate": "linear"},
      "encoding": {
        "x":{
          "field": "${field:step}",
          "type": "quantitative"
        },
        "y": {
          "field": "${field:lineVal}",
          "title": "y",
          "type": "quantitative"
        },
        "color": {
          "type": "nominal",
          "field": "name",
          "scale" : {
            "range" : {"field": "color"}
            }
        },
        "strokeDash": {
          "type": "nominal",
          "field": "${field:lineKey}"
        }
      }
    }
  ]
}```

### CODE REVIEW

**Review by staceysv — COMMENTED** [review]
looks great, excited to get this in! left two minor comments and one that might be a longer question (can we gracefully handle variable lengths of arrays across x, y1, y2, y3... on the Vega side? I don't think we should block on this, but if it's a small refactor/not a problem on the Vega side, I would love to get it in)

**Review by staceysv — APPROVED** [review]

File: wandb/plot/line_series.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        ys = {
+            "y1": [i for i in range(1,10)],
+            "y2": [i**2 for i in range(1,10)]
+        }
+        wandb.log({'line-series-plot1': wandb.plot.line_series(x, ys, "title", "step")})
+        ```
+    """
+
+    for _, value in ys.items():
+        assert len(value) == len(x), "All y arrays must match with length of x"
```
[the remark this answers is not in the document]
Comment (staceysv) [reply wandb/plot/line_series.py]: sounds like we can come back to this if need be :)

[the remark this answers is not in the document]
Comment (KyleGoyette) [reply wandb/plot/line_series.py]: This is fixed in the new code. Now I zip the two lists together and the input is xs =[[x11, x12,...], [x21, x22,...]] and ys= [[y11, y12,...], [y21, y22...]]

This is how we want it right?

**Comment by raubitsj** [comment]
@KyleGoyette , is this PR ready to be merged?

**Comment by KyleGoyette** [comment]
One more change I think, will have it done and reviewed today.

File: wandb/plot/line_series.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+from six.moves.collections_abc import Sequence
+import wandb
+
+
+def line_series(xs, ys, keys=None, title=None, xname=None):
+    """
+    Construct a line series plot.
+
+    Arguments:
+        xs (array of array): Array of arrays of x values
```
Comment (staceysv) [inline wandb/plot/line_series.py:10]: array of arrays

File: wandb/plot/line_series.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        ys (array of arrays): Array of y values
+        title (string): Plot title.
+        xname: Title of x-axis
+
+    Returns:
+        A plot object, to be passed to wandb.log()
+
+    Example:
+        ```
+        When logging a singlular array for x, all ys are plotted against that x
```
Comment (staceysv) [inline wandb/plot/line_series.py:20]: yes! great! small typo, singular

**Review by staceysv — APPROVED** [review]
looks great, love the examples

**Pull request merged**